include/gz/math/SphericalCoordinates.hh

```diff
@@ -141,6 +141,24 @@ namespace gz
                                      const gz::math::Angle &_latB,
                                      const gz::math::Angle &_lonB);
 
+
+      /// \brief Get the distance between two points expressed in geographic
+      /// latitude and longitude. It assumes that both points are at the
+      /// given radius from the centre of the celestial body.
+      /// Example: _latA = 38.0016667 and _lonA = -123.0016667) represents
+      /// the point with latitude 38d 0'6.00"N and longitude 123d 0'6.00"W.
+      /// \param[in] _latA Latitude of point A.
+      /// \param[in] _lonA Longitude of point A.
+      /// \param[in] _latB Latitude of point B.
+      /// \param[in] _lonB Longitude of point B.
+      /// \param[in] _radius Radius of the celestial body in metres.
+      /// \return Distance in meters.
+      public: static double Distance(const ignition::math::Angle &_latA,
+                                     const ignition::math::Angle &_lonA,
+                                     const ignition::math::Angle &_latB,
+                                     const ignition::math::Angle &_lonB,
+                                     const double &_radius);
```

Comment:
I think there are other functions which are currently hardcoded to the Earth and may eventually need to support other bodies.

The way that `SphericalCoordinates` exposes support for other bodies is through the `SurfaceType` enum. With that in mind, instead of having the user pass surface-specific parameters to each function, I think we should support other surface types. 

For your use case, I'd suggest either adding the surface type you need (maybe `MOON_SCS`) to the enum, or maybe adding a `CUSTOM` surface type. How I think those 2 options could work:

* Keep an internal `radius` variable, and initialize it to `g_EarthRadius` if the type is `EARTH_WGS84`, and do the same for other known types.
* Add a `double Radius() const` getter that returns that radius
* Add a `void SetRadius(double _radius)` setter to be used in case the surface type is `CUSTOM`
* Consider printing or returning errors if the user tries to call functions for `CUSTOM` without setting a radius first

```
auto moonSc = SphericalCoordinates(SurfaceType::MOON_SCS);
ASSERT_GT(moonSc.Radius(), 0);
ASSERT_GE(moonSc.Distance(latA, lonA, latB, lonB), 0);
```

```
auto moonSc = SphericalCoordinates(SurfaceType::CUSTOM);

// Radius not set for custom
ASSERT_LT(customSc.Radius(), 0);
ASSERT_LT(customSc.Distance(latA, lonA, latB, lonB), 0);;

// With custom radius
customSc.SetRadius(myRadius);

ASSERT_GT(customSc.Radius(), 0);
ASSERT_GE(customSc.Distance(latA, lonA, latB, lonB), 0);
```

Reply:
The method ``Distance()`` is currently a static method, so is it not bound to an instance. Setting the surface type will require a new non static method. Is that reasonable to add ?

Reply:
Based on an offline discussion with @chapulina , I'll write a new non static method for calculating distance, that takes into account the surface type.